src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -816,14 +816,14 @@ internal bool MethodGroupIsCompatibleWithDelegate(BoundExpression receiverOpt, b
                 return false;
             }
 
-            CompoundUseSiteInfo<AssemblySymbol> useSiteInfo = default;
+            CompoundUseSiteInfo<AssemblySymbol> useSiteInfo = GetNewCompoundUseSiteInfo(diagnostics);
 
             // If this is an extension method delegate, the caller should have verified the
             // receiver is compatible with the "this" parameter of the extension method.
             Debug.Assert(!isExtensionMethod ||
                 (Conversions.ConvertExtensionMethodThisArg(methodParameters[0].Type, receiverOpt.Type, ref useSiteInfo).Exists && useSiteInfo.Diagnostics.IsNullOrEmpty()));
 
-            useSiteInfo = default;
+            useSiteInfo = new CompoundUseSiteInfo<AssemblySymbol>(useSiteInfo);
```

Comment:
This pattern, with a constructor that takes a `CompoundUseSiteInfo` and drops the diagnostics/dependencies, feels particularly confusing to me. Please use a named method for this instead, something like `WithNoAccumulatedInfo` perhaps.

Reply:
Please don't won't fix this. This is confusing to a reader.

---
In reply to: [369824393](https://github.com/dotnet/roslyn/pull/41117#discussion_r369824393) [](ancestors = 369824393)

Reply:
It is a common pattern when an a constructor takes an instance of the same type. See CSharpCompilationOptions, etc. 

---
In reply to: [369826186](https://github.com/dotnet/roslyn/pull/41117#discussion_r369826186) [](ancestors = 369826186,369824393)